### PR TITLE
v4l2-to-ndi: init at 0-unstable-2022-09-14

### PR DESCRIPTION
**Pull request by pinpox**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/v4/v4l2-to-ndi/package.nix [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  installPhase = ''
+    mkdir $out
+    cp -r build $out/bin
+  '';
+
+  meta = with lib; {
+    description = "A video input (V4L2) to NDI converter";
+    homepage = "https://github.com/lplassman/V4L2-to-NDI";
+    license = licenses.mit;
+    maintainers = with pkgs; [ pinpox MayNiklas ];
```
Comment (Aleksanaa) [inline pkgs/by-name/v4/v4l2-to-ndi/package.nix:50]:
```suggestion
    maintainers = with maintainers; [ pinpox MayNiklas ];
```

**Comment by Aleksanaa** [comment]
Sorry for closing, CI not passing and without proper fix for a long time. If you want to work on this again, you can reopen it.

**Comment by pinpox** [comment]
@Aleksanaa Sorry, kinda forgot about this one. It seems to only be a simple fix though, so I just pushed a commit to fix it. If that was the only thing missing, I'd still be interested in having this in nixpkgs since I might use it in the future.

File: pkgs/by-name/v4/v4l2-to-ndi/package.nix (hunk 1 above)
Comment (pinpox) [reply pkgs/by-name/v4/v4l2-to-ndi/package.nix]: Fixed

File: pkgs/by-name/v4/v4l2-to-ndi/package.nix [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{ lib
+, stdenv
```
Comment (FliegendeWurst) [inline pkgs/by-name/v4/v4l2-to-ndi/package.nix:2]: Please format with the new formatter

File: pkgs/by-name/v4/v4l2-to-ndi/package.nix [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    cp -r build $out/bin
+  '';
+
+  meta = with lib; {
+    description = "A video input (V4L2) to NDI converter";
+    homepage = "https://github.com/lplassman/V4L2-to-NDI";
+    license = licenses.mit;
+    maintainers = with maintainers; [ pinpox MayNiklas ];
+    mainProgram = "v4l2-to-ndi";
+    platforms = platforms.linux;
```
Comment (FliegendeWurst) [inline pkgs/by-name/v4/v4l2-to-ndi/package.nix:52]:
```suggestion
  meta = {
    description = "Video input (V4L2) to NDI converter";
    homepage = "https://github.com/lplassman/V4L2-to-NDI";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ pinpox MayNiklas ];
    mainProgram = "v4l2-to-ndi";
    platforms = lib.platforms.linux;
```

File: pkgs/by-name/v4/v4l2-to-ndi/package.nix [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    -I'NDI SDK for Linux'/include/ \
+    -Iinclude/ \
+    -L'NDI SDK for Linux'/lib/x86_64-linux-gnu \
+    -o build/v4l2ndi main.cpp PixelFormatConverter.cpp -lndi -ldl
+  '';
+
+  installPhase = ''
+    mkdir $out
+    cp -r build $out/bin
+  '';
```
Comment (FliegendeWurst) [inline pkgs/by-name/v4/v4l2-to-ndi/package.nix:44]:
```suggestion
  installPhase = ''
    runHook preInstall
    mkdir $out
    cp -r build $out/bin
    runHook postInstall
  '';
```

File: pkgs/by-name/v4/v4l2-to-ndi/package.nix [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+
+  buildPhase = ''
+    mkdir build
+    g++ -std=c++14 -pthread  -Wl,--allow-shlib-undefined -Wl,--as-needed \
+    -I'NDI SDK for Linux'/include/ \
+    -Iinclude/ \
+    -L'NDI SDK for Linux'/lib/x86_64-linux-gnu \
+    -o build/v4l2ndi main.cpp PixelFormatConverter.cpp -lndi -ldl
+  '';
```
Comment (FliegendeWurst) [inline pkgs/by-name/v4/v4l2-to-ndi/package.nix:39]:
```suggestion
  buildPhase = ''
    runHook preBuild
    mkdir build
    g++ -std=c++14 -pthread  -Wl,--allow-shlib-undefined -Wl,--as-needed \
    -I'NDI SDK for Linux'/include/ \
    -Iinclude/ \
    -L'NDI SDK for Linux'/lib/x86_64-linux-gnu \
    -o build/v4l2ndi main.cpp PixelFormatConverter.cpp -lndi -ldl
    runHook postBuild
  '';
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by pinpox** [comment]
@FliegendeWurst applied suggestions, rebuilds fine. Also formatted and squashed

File: pkgs/by-name/v4/v4l2-to-ndi/package.nix [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+  installPhase = ''
+    runHook preInstall
+    mkdir $out
+    cp -r build $out/bin
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A video input (V4L2) to NDI converter";
```
Comment (FliegendeWurst) [inline pkgs/by-name/v4/v4l2-to-ndi/package.nix:51]:
```suggestion
    description = "Video input (V4L2) to NDI converter";
```

File: pkgs/by-name/v4/v4l2-to-ndi/package.nix [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  autoPatchelfHook,
+  openssl,
+  curl,
+  avahi,
+  ndi,
+}:
+
+stdenv.mkDerivation {
+  pname = "v4l2-to-ndi";
+  version = "unstable-2022-09-14";
```
Comment (FliegendeWurst) [inline pkgs/by-name/v4/v4l2-to-ndi/package.nix:14]:
```suggestion
  version = "0-unstable-2022-09-14";
```
I missed this last time.

File: pkgs/by-name/v4/v4l2-to-ndi/package.nix [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+  meta = with lib; {
+    description = "Video input (V4L2) to NDI converter";
+    homepage = "https://github.com/lplassman/V4L2-to-NDI";
+    license = licenses.mit;
+    maintainers = with maintainers; [
+      pinpox
+      MayNiklas
+    ];
+    mainProgram = "v4l2-to-ndi";
```
Comment (FliegendeWurst) [inline pkgs/by-name/v4/v4l2-to-ndi/package.nix:58]:
```suggestion
    mainProgram = "v4l2ndi";
```
Or you could fix the output name in line 39

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**